### PR TITLE
Add Vectorlike

### LINKED ISSUE #166: Accept tuples and vectors
**Is your feature request related to a problem? Please describe.**
Using vectors everywhere is really annoying. 

**Describe the solution you'd like**
Initializers should accept vectors and tuples. Functions that don't require vector math should also accept both.

### PR DESCRIPTION
**Pull request by tinmarr**
Allows for vectorlikes

Closes #166 
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [x] Documentation is up to date
  - [x] Relevent tutorials are up to date
  - [x] Changelog is up to date
  - [x] Test coverage is at 100%
  - [x] All checks are passing
  - [x] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*

### CODE REVIEW

File: rubato/c_src/c_draw.py [hunk 1]
```diff
@@ -85,7 +85,8 @@ def draw_poly(
     vxt = []
     vyt = []
     for v in points:
-        x, y = v.tuple_int()
+        Vector.raise_vector_like(v, "points")
+        x, y = (int(v[0]), int(v[1]))
```
Comment (mrElnekave) [inline rubato/c_src/c_draw.py:89]: get rid of parentheses

File: rubato/struct/gameobject/sprites/animation.py [hunk 2]
```diff
@@ -219,9 +223,11 @@ def add_spritesheet(
                 animation.add_spritesheet("idle", spritesheet, to_coord=spritesheet.end)
                 # This will just load from the start to the end of the spritesheet.
         """
+        Vector.raise_vector_like(from_coord, "from_coord")
+        Vector.raise_vector_like(to_coord, "to_coord")
         state = []
-        x, y = from_coord.tuple_int()
-        to_x, to_y = to_coord.tuple_int()
+        x, y = (int(from_coord[0]), int(from_coord[1]))
+        to_x, to_y = (int(to_coord[0]), int(to_coord[1]))
```
Comment (mrElnekave) [inline rubato/struct/gameobject/sprites/animation.py:230]: .

**Review by mrElnekave — APPROVED** [review]
looks good.

**Review by tsedan — CHANGES_REQUESTED** [review]
offset_demo is currently broken. will rereview once @tinmarr fixes

**Review by tsedan — APPROVED** [review]
Looks good!!! :)

**Pull request merged**